File: Models/ProjectType.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using DataPlane.Interfaces;

namespace DataPlane.Models
{
    public class ProjectType : BaseModel, IBaseEntity
    {
        // Primary key
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }
        public string Description { get; set; }
        public string TypeCode { get; set; }

        // Navigation property 
        public virtual ICollection<Project> Projects { get; } = new List<Project>();

    }
}

File: ClientApp/src/app/models/project-type.ts
import { Basemodel } from './basemodel';

export interface ProjectType {

    id: number;
    workspaceId: number;

    name: string;
    description: string;
    typeCode: string;

}

export class ProjectType extends Basemodel {

    public id: number;
    public workspaceId: number;

    public name: string;
    public description: string;
    public typeCode: string;

}
